packages/launchpad/src/components/ConfigFile.vue

```diff
@@ -59,51 +65,72 @@ fragment ConfigFile on Wizard {
 }
 `
 
-export default defineComponent({
-  components: {
-    WizardLayout,
-    PrismJs,
-    CopyButton,
-  },
-  props: {
-    gql: {
-      type: Object as PropType<ConfigFileFragment>,
-      required: true
-    }
-  },
-  setup(props) {
-    const manualInstall = ref(false);
+gql`
+fragment ProjectRoot on App {
+  activeProject {
+    projectRoot
+  }
+}
+`
 
-    const altFn = (val: boolean) => {
-      manualInstall.value = val
+gql`
+mutation appCreateConfigFile($code: String!, $configFilename: String!) {
+  appCreateConfigFile(code: $code, configFilename: $configFilename) {
+    activeProject {
+      projectRoot
     }
+  }
+}
+`
+
+const props = defineProps<{
+  wizard: ConfigFileFragment 
+  app: ProjectRootFragment | undefined
+}>()
+
+
+const manualInstall = ref(false);
 
-    const tsInstalled = ref(false);
-    const language = ref<"js" | "ts">("ts");
-    const nextButtonName = computed(() =>
-      manualInstall.value ? "I've added this file" : "Create File"
-    );
-
-    import("prismjs/components/prism-typescript").then(() => {
-      tsInstalled.value = true;
-    });
-
-    return {
-      manualInstall,
-      altFn,
-      nextButtonName,
-      code: computed(() => {
-        if (language.value === 'js') {
-          return props.gql.sampleCodeJs
-        }
-        return props.gql.sampleCodeTs
-      }),
-      language,
-      languages,
-      tsInstalled,
-    };
-  },
+const altFn = (val: boolean) => {
+  manualInstall.value = val
+}
+
+const tsInstalled = ref(false);
+const language = ref<"js" | "ts">("ts");
+const nextButtonName = computed(() =>
+  manualInstall.value ? "I've added this file" : "Create File"
+);
+
+import("prismjs/components/prism-typescript").then(() => {
+  tsInstalled.value = true;
 });
+
+
+const createConfigFile = useMutation(AppCreateConfigFileDocument)
+
+const createConfig = async () => {
+  console.log('Creating!')
+  if (!props.app?.activeProject?.projectRoot) {
+    throw Error(`Expected active project to be set with a projectRoot. This should never happen.`)
+  }
+
+  if (!code.value) {
+    // should be impossible 
+    throw Error(`Code is required to create a config file. Got ${code.value}.`)
+  }
+
+  await createConfigFile.executeMutation({ 
+    code: code.value,
+    configFilename: `cypress.config.${language.value}`
+  })
+}
+
+const code = computed(() => {
+  if (language.value === 'js') {
+    return props.wizard.sampleCodeJs
+  }
+  return props.wizard.sampleCodeTs
+})
```

Comment:
Can you put this above the usage? I was struggling to figure out why this wasn't undefined.